### ISSUE TITLE
Using GZ to transport large GeoJSON files

### ISSUE THREAD

**Issue by jerryhall** [tangrams/tangram#528]
I understand that one can transport archived (.gz) files, which should be parsed on the browser-side for sites that are so enabled. I believe I've enabled my site with the .htaccess settings from [https://philipgwynne.com/blog/2016-07-25-enabling-caching-and-gzip-compression-with-htaccess](url) then changed by scene file for a data source of 20k+ points. But, when the site loads I keep getting an error reading: 

"Tangram v0.12.0 [warn]: data source load error(s) for source '_licenses', URL 'https://path/to/ca_la_current.gz': SyntaxError: Unexpected token  in JSON at position 0"

Am I actually able to call such a zipped file to load a source from my scene and if so is this error because I'm calling the zipped file wrong or is my browser misinterpreting it? 

This could be a significant solution as my data layer would go from 19mb to 3mb as a zipped file. 

Thanks!

In my scene file I'm calling (properly indented): 
    _licenses:  
type: GeoJSON  
url: https://path/to/data.gz   and have tried https://path/to/data.geojson.gz

I've confirmed my site accepts zipped .gz files via https://varvy.com/tools/gzip/

**Comment by nvkelso** [comment]
What's the file extension after you decompress it?

**Comment by jerryhall** [comment]
It appears to remain as a .gz file, which I'm only discerning from the error message it returns. Is there a better way to confirm that?

**Comment by nvkelso** [comment]
Is it at a public web address we can try? (You can email to nathaniel@mapzen.com alternatively to posting here.)

**Comment by jerryhall** [comment]
Thanks for checking this out. I just emailed you all the info. I'll have a public site up asap and will share a lot of this work.

**Comment by bcamper** [comment]
It should work if your file is served with the `Content-Encoding: gzip` header.

**Comment by pnorman** [comment]
Generally if you're using content-encoding for protocol level compression you won't have a suffix of .gz. The **content** is uncompressed, but the transmission of the content is compressed.

You do **not** want to send gzip files with content-encoding, because you'd then be applying compression twice.

**Comment by jerryhall** [comment]
Thank you all. I didn't get to pnorman's comment until just now. Since I was using the .gz in the scene file and having a .gz file on the server it was indeed conflicting. I now host this on a CPanel-enabled host and I went to the Optimize Website link. I added the json and geojson elements to the end so the entire 'Compress the specified MIME types.' selection reads:   
text/html text/plain text/xml application/json application/vnd.geo+json

So, what's effectively happening is any time someone calls a page with json or geojson content, Apache serves it to the client zipped and the browser automatically unzips it.

This is a significant time and bandwidth saving tactic and it's been about eight months of my hacking away at this until I found this out. Thanks all.
